src/authentication/Login/LoginRequest.tsx

```
@@ -0,0 +1,39 @@
+import storage from '../../app/localStorage';
+
+const LoginRequest = async (username:string, password:string) => {
+let response = {
+error: '',
+};
+if (username.trim().length !== 0 && password.trim().length !== 0) {
+    try {
+        const result = await fetch('openmrs/ws/rest/v1/session', {
+        headers: {
+            Authorization: 'Basic ' + btoa(username + ':' + password),
+        },
+        method: 'GET',
+        redirect: 'follow',
+        });
+        const [headers, body] = await Promise.all([
+        result.headers,
+        result.json(),
+        ]);
+        if (body.authenticated) {
+            storage.saveInfo(body);
+            localStorage.setItem('authenticated', body.authenticated);
+            localStorage.setItem('Banner', 'true');
+            window.location.href = "/"
+        } 
+        else {
+            response.error = 'Invalid username or password';
+        }
+} 
+    catch (e) {
+        response.error = 'Network error';
```

This is a catch-all error. The above request can fail due to different reasons e.g 504 gateway timeout or server error with status code 500.  Please return the actual error and handle different status codes

---

Resolved.